src/components/Search/index.tsx

```diff
@@ -296,7 +296,7 @@ function Search({queryJSON, currentSearchResults, lastNonEmptySearchResults, onS
             const isFromSelfDM = item.reportID === CONST.REPORT.UNREPORTED_REPORTID;
             const isTransactionItem = isTransactionListItemType(item);
 
-            let reportID = isTransactionItem && (!item.isFromOneTransactionReport || isFromSelfDM) ? item.transactionThreadReportID : item.reportID;
+            let reportID = isTransactionItem && (!item.isFromOneTransactionReport || isFromSelfDM) && item.transactionThreadReportID !== '0' ? item.transactionThreadReportID : item.reportID;
```

Comment:
Can we use `UNREPORTED_REPORTID` for '0' value?

Reply:
Ofc, I wasn't aware of it but its much better